### PR TITLE
Support logit bias for OpenAI API

### PR DESCRIPTION
**Pull request by dylanwhawk**
This PR leverages PR #1469 to implement [logit bias](https://help.openai.com/en/articles/5247780-using-logit-bias-to-alter-token-probability-with-the-openai-api) for completions/chat completions in the OpenAI entry point

### CODE REVIEW

File: vllm/entrypoints/openai/protocol.py [hunk 1]
```diff
@@ -137,6 +139,21 @@ class CompletionRequest(BaseModel):
...
+        logits_processors = None
+
+        if self.logit_bias:
+
+            def logit_bias_logits_processor(
+                    token_ids: List[int],
+                    logits: torch.Tensor) -> torch.Tensor:
+                for token_id, bias in self.logit_bias.items():
+                    bias = min(100, max(
+                        -100, bias))  # Clamp the bias between -100 and 100
```
Comment (mgoin) [inline vllm/entrypoints/openai/protocol.py:151]: nit: maybe worth noting this is from the openai spec

**Review by mgoin — APPROVED** [review]

File: vllm/entrypoints/openai/protocol.py [hunk 2]
```diff
@@ -137,6 +139,21 @@ class CompletionRequest(BaseModel):
     def to_sampling_params(self):
         echo_without_generation = self.echo and self.max_tokens == 0
 
+        logits_processors = None
```
Comment (esmeetu) [inline vllm/entrypoints/openai/protocol.py:142]: Add these in `ChatCompletionRequest` which also supports.

File: vllm/entrypoints/openai/protocol.py [hunk 3]
```diff
@@ -137,6 +139,21 @@ class CompletionRequest(BaseModel):
...
+            def logit_bias_logits_processor(
+                    token_ids: List[int],
+                    logits: torch.Tensor) -> torch.Tensor:
+                for token_id, bias in self.logit_bias.items():
+                    # Clamp the bias between -100 and 100 per OpenAI API spec
+                    bias = min(100, max(-100, bias))
+                    logits[int(token_id)] += bias
+                return logits
+
+            logits_processors = [logit_bias_logits_processor]
```
Comment (esmeetu) [inline vllm/entrypoints/openai/protocol.py:155]: Is the bais logic processor mutually exclusive with the one passed in as a parameter by the user? For example, does it support being used together with the JSON logic processor?

**Comment by dylanwhawk** [comment]
@esmeetu I just added to `ChatCompletionRequest` and resolved conflicts

**Comment by esmeetu** [comment]
@dylanwhawk LGTM! Thanks for adding this feature!

**Review by esmeetu — APPROVED** [review]

**Comment by yuchenlin** [comment]
Thanks! Is there an example where we can use the feature in non API mode (i.e., the regular python function call for llm.generate)? Thanks!

**Pull request merged**